src/main/java/bowling/domain/NormalFrame.java

```diff
@@ -27,4 +29,50 @@ public FrameStrategy nextFrame(int frameNumber) {
         next = new NormalFrame();
         return next;
     }
+
+    @Override
+    public int score() {
+        if (hasNext()) {
+            return CALCULATION_NOT_COMPLETED;
+        }
+        return FrameResult.scoreByResult(this, result(size()));
+    }
+
+    public int scoreInStrike() {
+        if (next.hasNext() && next.pinNumbers.size() < REQUIRED_PIN_COUNT) {
+            return CALCULATION_NOT_COMPLETED;
+        }
+        if (next.pinNumbers.index(0).isStrike()) {
+            return scoreInDouble();
+        }
+        return pinNumbers.sum() + next.pinNumbers.sum();
+    }
+
+    private int scoreInDouble() {
+        if (next.next == null) {
+            return scoreByFinalFrame();
+        }
+        if (next.next.pinNumbers.size() == EMPTY_FRAME) {
+            return CALCULATION_NOT_COMPLETED;
+        }
+        return pinNumbers.sum() + next.pinNumbers.sum() + next.next.pinNumbers.index(0).pinNumber();
+    }
+
+    private int scoreByFinalFrame() {
+        if (next.pinNumbers.size() < REQUIRED_PIN_COUNT) {
+            return CALCULATION_NOT_COMPLETED;
+        }
+        return pinNumbers.sum() + next.pinNumbers.index(0).pinNumber() + next.pinNumbers.index(1).pinNumber();
+    }
+
+    public int scoreInSpare() {
+        if (next.pinNumbers.size() == EMPTY_FRAME) {
+            return CALCULATION_NOT_COMPLETED;
+        }
+        return pinNumbers.sum() + next.pinNumbers.index(0).pinNumber();
```

Comment:
```suggestion
        return pinNumbers.sum() + next.firstScore();
```
이런식으로 표현할 수 있도록 만들면 어떨까요?